### PR TITLE
feature: get namespace name from release command

### LINKED ISSUE #1498: Support for dynamic namespace name in the Helm chart
#### What would you like to be added:
A namespace field in the helm values.yml

#### Why is this needed:
It is not only a very common setting but it could be mandatory in enterprises environment where namespaces have naming conventions restrictions like `$env-$team-security`

#### User story covered
Not sure.

I will try to open a PR soon if no one stops me.

PS: I'm new to the community and I'm trying to respect the huge documentation about contributing, please tell me if anything is wrong.

### PR DESCRIPTION
**Pull request by tuxerrante**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
It is an important missing setting but it also could be mandatory in enterprises environment where namespaces have naming conventions.

#### Which issue(s) this PR fixes:
#1498 

Fixes #1498 

#### Does this PR have test?
```bash
on ⛵ aks-test () security-profiles-operator go v1.19.5 took 5s
[15:13:52] ❯ helm install --create-namespace  --namespace tuxerrante-security security-profiles-operator deploy/helm/
WARNING: Kubernetes configuration file is group-readable. This is insecure. Location: /mnt/c/Users/alessandro.affinito/.kube/config
WARNING: Kubernetes configuration file is world-readable. This is insecure. Location: /mnt/c/Users/alessandro.affinito/.kube/config
W0223 15:14:16.478807    2097 warnings.go:70] spec.template.metadata.annotations[seccomp.security.alpha.kubernetes.io/pod]: deprecated since v1.19, non-functional in a future release; use the "seccompProfile" field instead
NAME: security-profiles-operator
LAST DEPLOYED: Thu Feb 23 15:14:11 2023
NAMESPACE: tuxerrante-security
STATUS: deployed
REVISION: 1
TEST SUITE: None

[15:24:01] ❯ kubectl get pods -n tuxerrante-security
NAME                                                  READY   STATUS    RESTARTS   AGE
security-profiles-operator-844954996d-9jnpk           1/1     Running   0          9m48s
security-profiles-operator-844954996d-gbrhh           1/1     Running   0          9m48s
security-profiles-operator-844954996d-rjbkj           1/1     Running   0          9m48s
security-profiles-operator-webhook-6b9c98b975-m2ccq   1/1     Running   0          2m45s
security-profiles-operator-webhook-6b9c98b975-szlts   1/1     Running   0          16m
security-profiles-operator-webhook-6b9c98b975-zr5ql   1/1     Running   0          16m
spod-5hszj                                            2/2     Running   0          9m29s
spod-qz74b                                            2/2     Running   0          9m29s


[15:24:59] ➜  kubectl logs -n tuxerrante-security deployments/security-profiles-operator
Found 3 pods, using pod/security-profiles-operator-844954996d-9jnpk
I0223 14:14:17.969898       1 main.go:258]  "msg"="Set logging verbosity to 0"
I0223 14:14:17.969940       1 main.go:264]  "msg"="Profiling support enabled: false"
I0223 14:14:17.969996       1 main.go:284] setup "msg"="starting component: security-profiles-operator" "buildDate"="1980-01-01T00:00:00Z" "buildTags"="netgo,osusergo,seccomp,apparmor" "cgoldFlags"="unknown" "compiler"="gc" "dependencies"="github.com/acobaugh/osrelease v0.1.0 ,github.com/aquasecurity/libbpfgo v0.4.5-libbpf-1.0.1 ,github.com/beorn7/perks v1.0.1 ,github.com/blang/semver/v4 v4.0.0 ,github.com/cert-manager/cert-manager v1.10.1 ,github.com/cespare/xxhash/v2 v2.2.0 ,github.com/containers/common v0.51.0 ,github.com/cpuguy83/go-md2man/v2 v2.0.2 ,github.com/davecgh/go-spew v1.1.1 ,github.com/emicklei/go-restful/v3 v3.8.0 ,github.com/evanphx/json-patch/v5 v5.6.0 ,github.com/fsnotify/fsnotify v1.6.0 ,github.com/go-logr/logr v1.2.3 ,github.com/go-openapi/jsonpointer v0.19.5 ,github.com/go-openapi/jsonreference v0.20.0 ,github.com/go-openapi/swag v0.22.3 ,github.com/gogo/protobuf v1.3.2 ,github.com/golang/groupcache v0.0.0-20210331224755-41bb18bfe9da ,github.com/golang/protobuf v1.5.2 ,github.com/google/gnostic v0.6.9 ,github.com/google/go-cmp v0.5.9 ,github.com/google/gofuzz v1.2.0 ,github.com/google/uuid v1.3.0 ,github.com/imdario/mergo v0.3.13 ,github.com/jellydator/ttlcache/v3 v3.0.1 ,github.com/josharian/intern v1.0.0 ,github.com/json-iterator/go v1.1.12 ,github.com/mailru/easyjson v0.7.7 ,github.com/matttproud/golang_protobuf_extensions v1.0.2-0.20181231171920-c182affec369 ,github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd ,github.com/modern-go/reflect2 v1.0.2 ,github.com/munnerz/goautoneg v0.0.0-20191010083416-a7dc8b61c822 ,github.com/nxadm/tail v1.4.8 ,github.com/opencontainers/runtime-spec v1.0.3-0.20220825212826-86290f6a00fb ,github.com/openshift/api v0.0.0-20221205111557-f2fbb1d1cd5e ,github.com/pjbgf/go-apparmor v0.1.1 ,github.com/pkg/errors v0.9.1 ,github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring v0.62.0 ,github.com/prometheus/client_golang v1.14.0 ,github.com/prometheus/client_model v0.3.0 ,github.com/prometheus/common v0.37.0 ,github.com/prometheus/procfs v0.8.0 ,github.com/russross/blackfriday/v2 v2.1.0 ,github.com/seccomp/libseccomp-golang v0.10.0 ,github.com/sirupsen/logrus v1.9.0 ,github.com/spf13/pflag v1.0.5 ,github.com/urfave/cli/v2 v2.24.4 ,github.com/xrash/smetrics v0.0.0-20201216005158-039620a65673 ,golang.org/x/mod v0.8.0 ,golang.org/x/net v0.7.0 ,golang.org/x/oauth2 v0.4.0 ,golang.org/x/sync v0.1.0 ,golang.org/x/sys v0.5.0 ,golang.org/x/term v0.5.0 ,golang.org/x/text v0.7.0 ,golang.org/x/time v0.0.0-20220609170525-579cf78fd858 ,gomodules.xyz/jsonpatch/v2 v2.2.0 ,google.golang.org/genproto v0.0.0-20230110181048-76db0878b65f ,google.golang.org/grpc v1.53.0 ,google.golang.org/protobuf v1.28.1 ,gopkg.in/inf.v0 v0.9.1 ,gopkg.in/tomb.v1 v1.0.0-20141024135613-dd632973f1e7 ,gopkg.in/yaml.v2 v2.4.0 ,gopkg.in/yaml.v3 v3.0.1 ,k8s.io/api v0.25.4 ,k8s.io/apiextensions-apiserver v0.25.4 ,k8s.io/apimachinery v0.25.5 ,k8s.io/client-go v0.25.4 ,k8s.io/component-base v0.25.4 ,k8s.io/klog/v2 v2.90.0 ,k8s.io/kube-openapi v0.0.0-20220803164354-a70c9af30aea ,k8s.io/utils v0.0.0-20221108210102-8e77b1f39fe2 ,sigs.k8s.io/controller-runtime v0.13.1 ,sigs.k8s.io/gateway-api v0.5.0 ,sigs.k8s.io/json v0.0.0-20220713155537-f223a00ba0e2 ,sigs.k8s.io/release-utils v0.7.3 ,sigs.k8s.io/structured-merge-diff/v4 v4.2.3 ,sigs.k8s.io/yaml v1.3.0 " "gitCommit"="unknown" "gitCommitDate"="unknown" "gitTreeState"="clean" "goVersion"="go1.20.1" "ldFlags"="unknown" "libbpf"="v1.1" "libseccomp"="2.5.4" "platform"="linux/amd64" "version"="0.6.1-dev"
...
I0223 14:24:59.549265       1 spod_controller.go:271] spod-config "msg"="Adding 'Updating' status to the SPOD Instance" "namespace"="tuxerrante-security" "profile"="spod"
I0223 14:25:27.507891       1 ca.go:62] spod-config "msg"="Using cert-manager as certificate provider"
I0223 14:25:27.508218       1 spod_controller.go:391] spod-config "msg"="Updating cert manager resources"
I0223 14:25:27.578325       1 spod_controller.go:398] spod-config "msg"="Updating operator webhook"
I0223 14:25:27.616648       1 spod_controller.go:404] spod-config "msg"="Updating operator daemonset"
I0223 14:25:27.640967       1 spod_controller.go:409] spod-config "msg"="Updating operator default profiles"
I0223 14:25:27.640989       1 spod_controller.go:445] spod-config "msg"="Updating metrics service"
I0223 14:25:27.651314       1 spod_controller.go:450] spod-config "msg"="Updating operator service monitor"
I0223 14:25:28.701598       1 request.go:682] Waited for 1.041750808s due to client-side throttling, not priority and fairness, request: GET:https://10.0.0.1:443/apis/acme.cert-manager.io/v1?timeout=32s
I0223 14:25:29.209567       1 spod_controller.go:455] spod-config "msg"="Service monitor resource does not seem to exist, ignoring"
I0223 14:25:29.209601       1 spod_controller.go:271] spod-config "msg"="Adding 'Updating' status to the SPOD Instance" "namespace"="tuxerrante-security" "profile"="spod"
```
#### Does this PR introduce a user-facing change?

```release-note
CLI change: --namespace flag should be specified when installing the helm chart
```

It will install the app in the default namespace if no --namespace argument is given. Which is also the expected behaviour.

### CODE REVIEW

File: deploy/overlays/helm/kustomization.yaml
Comment (pjbgf) [inline deploy/overlays/helm/kustomization.yaml:20]: Let's do this by type instead:

```suggestion
patches:
- path: operator-profile.yaml
  target:
    kind: (ServiceAccount|Role|ClusterRoleBinding|RoleBinding|Secret)

- patch: |
    - op: replace
      path: /subjects/0/namespace
      value: "{{ .Release.Namespace }}"
  target:
    kind: (ClusterRoleBinding|RoleBinding)

- patch: |
    - op: replace
      path: /metadata/name
      value: "{{ .Release.Namespace }}"
  target:
    kind: Namespace
```

File: installation-usage.md
Comment (saschagrunert) [inline installation-usage.md:53]: Can we please revert those indentation changes?

Comment (tuxerrante) [reply installation-usage.md:53]: I wonder from where they came from, maybe some pre commit hook

File: installation-usage.md
Comment (tuxerrante) [inline installation-usage.md:172]: @saschagrunert @pjbgf  should this be updated to [v1.11.0](https://github.com/cert-manager/cert-manager/releases/tag/v1.11.0)?

Comment (pjbgf) [reply installation-usage.md:173]: I would that on a separate PR.

**Comment by tuxerrante** [comment]
@pjbgf it seems to me that helm deployments are actually tested with plain kubectl create command, so failing to parse parameters. 
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_security-profiles-operator/1499/pull-security-profiles-operator-test-e2e/1630566792629850112#

Maybe we could infer from some[ e2e struct](https://github.com/kubernetes-sigs/security-profiles-operator/blob/main/test/suite_test.go#L86) field if the test should be run with helm instead?

File: .gitattribute
Comment (pjbgf) [inline .gitattribute:1]: The `.gitattribute` can be set system-wide within your machine, I am not sure it makes sense to add it to the repo.

File: internal/pkg/config/config.go
Comment (pjbgf) [inline internal/pkg/config/config.go:32]: Not sure we need to add this.
```suggestion
```

**Review by pjbgf — CHANGES_REQUESTED** [review]
It generally looks good to me, apart from some smalls nits below.

Also, I am not sure we need the two new files:
```
hack/ci/Vagrantfile-ubuntu-local
hack/ci/vagrant_vars.sh
```

It would be great to get some more eyes on it as well (@kubernetes-sigs/security-profiles-operator-maintainers).

File: .gitattribute
Comment (tuxerrante) [reply .gitattribute:1]: No problem, I've found some resource suggesting to add it tothe repo [[1](https://dev.to/deadlybyte/please-add-gitattributes-to-your-git-repository-1jld), [2](https://stackoverflow.com/a/15108707/3673430), [3](https://rehansaeed.com/gitattributes-best-practices/)] but I'm ok with whatever reduce the impact on other devs.

File: internal/pkg/config/config.go
Comment (tuxerrante) [reply internal/pkg/config/config.go:32]: I think here some fixing is needed. 
I've found some referral in code using "OperatorName" as namespace name. This could be confusing and it is not documented

**Comment by tuxerrante** [comment]
> It generally looks good to me, apart from some smalls nits below.
> 
> Also, I am not sure we need the two new files:
> 
> ```
> hack/ci/Vagrantfile-ubuntu-local
> hack/ci/vagrant_vars.sh
> ```
> 
> It would be great to get some more eyes on it as well (@kubernetes-sigs/security-profiles-operator-maintainers).

You're right about vagrant_vars.sh, this is a place holder I've left to open a future issue: there are mulitple and redundant values hardcoded anywhere in config files and scripts, there should be one config file with all these values and all the other files should read by it (app image version, go version, every var defined in the Makefile..).

Instead I find Vagrantfile-ubuntu-local file very useful for local testing and avoiding multiple pushes just to wait ci jobs results. Maybe it could be renamed, moved or modified instead of deleted?

File: deploy/base/role_binding.yaml
Comment (pjbgf) [inline deploy/base/role_binding.yaml:6]: I think we should keep the `namespace` here, or this will break for non helm deployments.

File: deploy/overlays/helm/kustomization.yaml
Comment (pjbgf) [inline deploy/overlays/helm/kustomization.yaml:20]:
```suggestion
  # Replace the namespace into the role binding associations.
```

File: deploy/overlays/helm/kustomization.yaml
Comment (pjbgf) [inline deploy/overlays/helm/kustomization.yaml:69]:
```suggestion
    kind: (ClusterRole|ClusterRoleBinding|ConfigMap|MutatingWebhookConfiguration|Namespace|Role|RoleBinding|Secret|ServiceAccount)

```

File: installation-usage.md
Comment (pjbgf) [inline installation-usage.md:185]:
```suggestion
# Label and annotate the ns to make it manageable by helm. Ensure it is
# running on the privileged Pod Security Standard.
```

File: installation-usage.md
Comment (pjbgf) [inline installation-usage.md:210]:
```suggestion
To remove everything or to do a new installation from scratch be sure to remove them first.
```

**Review by pjbgf — COMMENTED** [review]
Just a few other nits.

Apart from that I am not sure [about the namespaces](https://github.com/kubernetes-sigs/security-profiles-operator/pull/1499/files#r1134611144) removal on `deploy/base/role_binding/yaml`.

File: deploy/base/role_binding.yaml
Comment (pjbgf) [reply deploy/base/role_binding.yaml:6]: all tests are passing so probably this isn't an issue.

**Comment by pjbgf** [comment]
Apart from the tiny nits, it looks good. Just adding hold for other folks to also review.

/lgtm
/hold

File: deploy/base/role_binding.yaml
Comment (tuxerrante) [reply deploy/base/role_binding.yaml:6]: I think it is correct in this way, these are non namespaced resources, you can check it on docs or with the command 
`kubectl api-resources --namespaced=true`

File: hack/ci/Vagrantfile-ubuntu-local
Comment (saschagrunert) [inline hack/ci/Vagrantfile-ubuntu-local:1]: Do we need this file?

Comment (JAORMX) [reply hack/ci/Vagrantfile-ubuntu-local:1]: It was originally introduced for him to do local testing. I think it could be a nice addition... but let's then add that to a dedicated commit/PR.

**Review by saschagrunert — APPROVED** [review]

**Review by JAORMX — APPROVED** [review]
/lgtm

thanks for working on this!

**Pull request merged**
